test/traversals/dfs.jl

```diff
@@ -20,7 +20,7 @@
 
     for g in testdigraphs(gx)
       @test @inferred(is_cyclic(g))
-      @test_throws ErrorException topological_sort_by_dfs(g)      
+      @test_throws Exception topological_sort_by_dfs(g)      
```

Comment:
Shouldn't we specify the exception that's produced?

Reply:
The error changed from a general error to a `CycleError`. These tests (the one in `test/traversals`) will be going away in 2.0 since the functions they're testing won't exist anymore.

Reply:
That said, it probably makes sense to do it.